mango-core/src/main/scala/org/bdgenomics/mango/models/TrackedLayout.scala

```diff
@@ -117,6 +141,33 @@ class OrderedTrackedLayout[T: ClassTag](values: Traversable[(ReferenceRegion, T)
     }
   }
 
+  private def findPairsAndAddToTrack(readPairs: Map[String, Seq[(ReferenceRegion, T)]]) {
+    readPairs.foreach {
+      p =>
+        {
+          val recs: Seq[(ReferenceRegion, T)] = p._2
+          val trackOption: Option[Track] = TrackTimers.FindConflict.time {
+            trackBuilder.find(track => !track.conflicts(recs))
+          }
+          val track: Track = trackOption.getOrElse(addTrack(new Track()))
+
+          recs.foreach {
+            rec: (ReferenceRegion, T) =>
+              {
+                if (!track.conflicts(rec)) {
+                  track += rec
+                  trackBuilder -= track
```

Comment:
It's a method of efficiently laying out the tracks. The idea is that when a record is added to a track, that track most likely won't have rooms for nearby additional records, so we move the track to the end of the ListBuffer. By constantly moving filled up tracks to the end, the tracks at the start of the ListBuffer will most likely have room for the current record we're looking at. Therefore, `trackBuilder.find(track => !track.conflicts(recs))` only has to look at the first few tracks in our ListBuffer before finding a nonconflicting track.